automation/vars/default.yaml

```diff
@@ -0,0 +1,46 @@
+---
+vas:
+  hci:
+    repo: https://github.com/openstack-k8s-operators/architecture.git
+    branch: main
+    stages:
+    - path: examples/va/hci/control-plane/nncp
+      validation: "oc wait nncp --for jsonpath='{.status.conditions[0].reason}'=SuccessfullyConfigured --timeout=60" 
+      values:
+      - name: network-values
+        src_file: network_values.yaml
+        dest_file: values.yaml
+
+    - path: examples/va/hci/control-plane
+      validation: "oc wait osctlplane controlplane --for condition=Ready --timeout=600s"
+      values:
+      - name: network-values
+        src_file: network_values.yaml
+        dest_file: nncp/values.yaml
+
+    - path: examples/va/hci/edpm-pre-ceph
+      validation: "oc wait osdpd edpm-deployment-pre-ceph --for condition=Ready --timeout=1200s"
+      values:
+      - name: edpm-values
+        src_file: edpm_pre_ceph_values.yaml
+        dest_file: values.yaml
+
+    # Install Ceph on the OSP compute nodes in this step. OpenStack
+    # K8S CRDs do not provide a way to install Ceph via any sort of
+    # combination of CRs.
+    - path: ""
```

Comment:
To get Ceph installation done, and to allow some more flexibility in a near future, I'd propose to add `pre_stage_run` and `post_stage_run` entries.

With this, we'd get something like:
```YAML
vas:
  hci:
    stages:
      - path: aaaa
        validation: foo bar baz
       post_stage_run:
         - name: Deploy Ceph
           type: playbook
           source: /path/to/playbooks/ceph.yml
           inventory: /path/to/zuul_inventory.yml
        values:
          - name: edpm-values
            src_file: ...
            dest_file: ....
```

And as for `pre_stage_run`, we may be able to query about Ceph secrets, unless we properly update the `ceph.yml` playbook to inject those data directly.

The structure of `pre_stage_run` and `post_stage_run` would be the same as the hooks mechanism nested in the ci-framework.
Note that it may be good to use some jinja2 parameters when it comes to hook paths (such as `source` and `inventory`) to allow the same flexibility we have in the framework regarding code location.

With that added feature, and some more tweaking to this PR while I'm integrating the automation in the framework, we should be on a safe path even for future VA (or, even DT, provided we'll provide automation for them as well).

Any thoughts?

Reply:
I would say this makes sense as long as we consider this to be framework-agnostic.  The current suggestion doesn't seem to be CIFMW-specific (even though the example provided contains values that obviously work for that context, but that's fine).

Reply:
When Ceph is deployed by the ci-framework and the following PR has merged, then you'll have a post-ceph values.yaml file. There should not be any need to mention Ceph secrets on the `automation/vars/default.yaml`. Perhaps just the source `/home/zuul/architecture/examples/va/hci/values.yaml` which is specific to the architecture repo (`cifmw_ceph_client_values_post_ceph_path_src`)

https://github.com/openstack-k8s-operators/ci-framework/pull/1130